src/ess/reduce/nexus.py

```diff
@@ -0,0 +1,345 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+
+"""NeXus utilities.
+
+This module defines functions and domain types that can be used
+to build Sciline pipelines for simple workflows.
+If multiple different kind sof files (e.g., sample and background runs)
+are needed, custom types and providers need to be defined to wrap
+the basic ones here.
+"""
+
+from contextlib import nullcontext
+from pathlib import Path
+from typing import BinaryIO, ContextManager, NewType, Optional, Type, Union, cast
+
+import scipp as sc
+import scippnexus as snx
+
+FilePath = NewType('FilePath', Path)
+"""Full path to a NeXus file on disk."""
+NeXusFile = NewType('NeXusFile', BinaryIO)
+"""An open NeXus file.
+
+Can be any file handle for reading binary data.
+
+Note that this cannot be used as a parameter in Sciline as there are no
+concrete implementations of ``BinaryIO``.
+The type alias is provided for callers of load functions outside of pipelines.
+"""
+NeXusGroup = NewType('NeXusGroup', snx.Group)
+"""A ScippNexus group in an open file."""
+
+DetectorName = NewType('DetectorName', str)
+"""Name of a detector (bank) in a NeXus file."""
+EntryName = NewType('EntryName', str)
+"""Name of an entry in a NeXus file."""
+MonitorName = NewType('MonitorName', str)
+"""Name of a monitor in a NeXus file."""
+SourceName = NewType('SourceName', str)
+"""Name of a source in a NeXus file."""
+
+RawDetector = NewType('RawDetector', sc.DataGroup)
+"""Full raw data from a NeXus detector."""
+RawDetectorData = NewType('RawDetectorData', sc.DataArray)
+"""Data extracted from a RawDetector."""
+RawMonitor = NewType('RawMonitor', sc.DataGroup)
+"""Full raw data from a NeXus monitor."""
+RawMonitorData = NewType('RawMonitorData', sc.DataArray)
+"""Data extracted from a RawMonitor."""
+RawSample = NewType('RawSample', sc.DataGroup)
+"""Raw data from a NeXus sample."""
+RawSource = NewType('RawSource', sc.DataGroup)
+"""Raw data from a NeXus source."""
+
+
+def load_detector(
+    file_path: Union[FilePath, NeXusFile, NeXusGroup],
```

Comment:
It works as long as any type is provided. So, this can be used in a pipeline as is. But I think all production pipelines will need different kinds of files. So we will have to wrap it.